include/proxsuite/proxqp/results.hpp

```diff
@@ -128,14 +131,31 @@ struct Results
     info.dua_res = 0.;
     info.status = QPSolverOutput::PROXQP_MAX_ITER_REACHED;
   }
-  void cold_start()
+  void cold_start(std::optional<T> rho = std::nullopt,
+                  std::optional<T> mu_eq = std::nullopt,
+                  std::optional<T> mu_in = std::nullopt)
   {
 
-    info.rho = 1e-6;
-    info.mu_eq_inv = 1e3;
-    info.mu_eq = 1e-3;
-    info.mu_in_inv = 1e1;
-    info.mu_in = 1e-1;
+    if (rho != std::nullopt) {
+      info.rho = rho.value();
+    } else {
+      info.rho = 1e-6;
+    }
+    if (mu_eq != std::nullopt) {
+      info.mu_eq = mu_eq.value();
+      info.mu_eq_inv = T(1) / info.mu_eq;
+    } else {
+      info.mu_eq_inv = 1e3;
+      info.mu_eq = 1e-3;
+    }
+    if (mu_in != std::nullopt) {
+      info.mu_in = mu_in.value();
+      info.mu_in_inv = T(1) / info.mu_in;
+    } else {
+      info.mu_in_inv = 1e1;
+      info.mu_in = 1e-1;
+    }
+
```

Comment:
Please remove the else, and first set the default values, which you should then erase with the new ones.

Reply:
@Bambade Did you handle it?

Reply:
Yes, it's here: https://github.com/Simple-Robotics/proxsuite/pull/26/files#diff-a6c8c233cb988a4c3df99b6531cce47679b472324653597e1b491ac19c528f84R141

Reply:
Perfect.